File: pinyougou_controller_user_337/src/main/java/com/itheima/core/controller/UserController.java
package com.itheima.core.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.itheima.core.pojo.user.User;
import com.itheima.core.service.UserService;
import com.itheima.core.utils.PhoneFormatCheckUtils;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * @描述: 用户管理包括:
 *        用户注册
 *        用户登录
 *        用户个人中心
 * @Auther: yanlong
 * @Date: 2019/3/22 14:11:16
 * @Version: 1.0
 */
@SuppressWarnings("all")
@RestController
@RequestMapping("/user")
public class UserController {

    @Reference
    private UserService userService;

    /**
     *@描述: 发送验证码
     *@Param: [phone]
     *@Return: entity.Result
    */

    @RequestMapping("/sendCode")
    public Result sendCode(String phone){
        //判断手机号是否合法
        if (PhoneFormatCheckUtils.isPhoneLegal(phone)){

            try {
                userService.sendCode(phone);
                return new Result(true,"发送成功");
            } catch (Exception e) {
                e.printStackTrace();
                return new Result(false,"发送失败");
            }
        }else {
            return new Result(false,"手机号不合法");
        }

    }


    /**
     *@描述: 新增用户
     *@Param: [smscode, user]
     *@Return: entity.Result
    */
    @RequestMapping("/add")
    public Result add(String smscode, @RequestBody User user){

            try {
                userService.add(smscode,user);
                return new Result(true,"注册成功");
            } catch (RuntimeException e) {
                return new Result(false,e.getMessage());//RuntimeException异常是从后面抛过来的,动态获取错误信息
            } catch (Exception e) {
                e.printStackTrace();
                return new Result(false,"注册失败");
            }
    }
}

File: pinyougou_controller_user_337/src/main/webapp/js/controller/userController.js
 //控制层 
app.controller('userController' ,function($scope,$controller   ,userService){	
	$scope.entity = {};
	//注册用户
	$scope.reg=function(){
		//判断用户不能为空
		if($scope.entity.username == null){
			alert("用户不能为空");
			return ;
		}

        if($scope.password == null || $scope.entity.password == null){
            alert("密码不能为空");
            return ;
        }

        //比较两次输入的密码是否一致
		if($scope.password!=$scope.entity.password){
			alert("两次输入密码不一致，请重新输入");
			$scope.entity.password="";
			$scope.password="";//确认密码
			return ;			
		}
		//新增
		userService.add($scope.entity,$scope.smscode).success(
			function(response){
				alert(response.message);
			}		
		);
	}
    
	//发送验证码
	$scope.sendCode=function(){
		if($scope.entity.phone==null || $scope.entity.phone==""){
			alert("请填写手机号码");
			return ;
		}
		
		userService.sendCode($scope.entity.phone  ).success(
			function(response){
				alert(response.message);
			}
		);		
	}
	
});	
